src/Api/Controllers/ReportsController.cs

```diff
@@ -0,0 +1,148 @@
+﻿using Api.Models.Response.Organizations;
+using Bit.Core.AdminConsole.Repositories;
+using Bit.Core.Context;
+using Bit.Core.Repositories;
+using Bit.Core.Vault.Queries;
+using Core.AdminConsole.OrganizationFeatures.OrganizationUsers.Interfaces;
+using Core.AdminConsole.OrganizationFeatures.OrganizationUsers.Requests;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Api.Controllers;
+
+public class CipherCounts
+{
+    public Guid CollectionId { get; set; }
+    public int ItemCount { get; set; }
+}
+
+[Route("reports")]
+[Authorize("Application")]
+public class ReportsController : Controller
+{
+    private readonly IOrganizationUserUserDetailsQuery _organizationUserUserDetailsQuery;
+    private readonly IGroupRepository _groupRepository;
+    private readonly ICollectionRepository _collectionRepository;
+    private readonly ICurrentContext _currentContext;
+    private readonly IOrganizationCiphersQuery _organizationCiphersQuery;
+    private readonly IOrganizationUserRepository _organizationUserRepository;
+
+    public ReportsController(
+        IOrganizationUserUserDetailsQuery organizationUserUserDetailsQuery,
+        IGroupRepository groupRepository,
+        ICollectionRepository collectionRepository,
+        ICurrentContext currentContext,
+        IOrganizationCiphersQuery organizationCiphersQuery,
+        IOrganizationUserRepository organizationUserRepository
+    )
+    {
+        _organizationUserUserDetailsQuery = organizationUserUserDetailsQuery;
+        _groupRepository = groupRepository;
+        _collectionRepository = collectionRepository;
+        _currentContext = currentContext;
+        _organizationCiphersQuery = organizationCiphersQuery;
+        _organizationUserRepository = organizationUserRepository;
+    }
+
+    [HttpGet("member-access/{orgId}")]
+    public async Task<IEnumerable<MemberAccessReportModel>> GetMemberAccessReport(Guid orgId)
```

Comment:
This is a lot of code in the controller - I know it's just a draft - but may be worth considering whether you can move any of this into constructors or a service/query method etc.